### PR TITLE
Topic fix server audio allocator maximum

### LINKED ISSUE #2758: TestBus:test_free failures
Is this failing for anyone else?

```
sclang> UnitTest.runTest("TestBus:test_free")

ERROR: Meta_Bus:audio: failed to get an audio bus allocated. numChannels: 1 server: localhost
ERROR: Meta_Bus:audio: failed to get an audio bus allocated. numChannels: 1 server: localhost
ERROR: Meta_Bus:audio: failed to get an audio bus allocated. numChannels: 1 server: localhost
ERROR: Meta_Bus:audio: failed to get an audio bus allocated. numChannels: 1 server: localhost
[huge number of copies of the above line omitted]

FAIL:a TestBus:test_free - after freeing, should be able to re-allocate all busses


FAIL:a TestBus:test_free -  should be numAudioBusChannels busses 
Is:
	 108 
Should be:
	 1020 



UNIT TEST.............
There were failures:
a TestBus:test_free - should be able to allocate all busses
a TestBus:test_free -  should be numAudioBusChannels busses 
Is:
	 108 
Should be:
	 1020 

a TestBus:test_free - after freeing, should be able to re-allocate all busses
a TestBus:test_free -  should be numAudioBusChannels busses 
Is:
	 108 
Should be:
	 1020 

```

### PR DESCRIPTION
**Pull request by telephon**
Because ContiguousBlockAllocator offsets the initial position from the
size, we need to account for offset in size explicitly.

This fixes #2758

### CODE REVIEW

File: testsuite/classlibrary/TestServerOptions.sc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+					options.numPrivateAudioBusChannels + options.numInputBusChannels + options.numOutputBusChannels
+				),
+				"sections of audio channels should add up to the total number of audio channels\nnumPrivateAudioBusChannels: %, numInputBusChannels: %, numOutputBusChannels: %\n"
+				.format(options.numPrivateAudioBusChannels, options.numInputBusChannels, options.numOutputBusChannels)
+			);
+
+		};
+
+		testAudio.value;
+		options.numAudioBusChannels = rrand(16, 1000);
```
Comment (snappizz) [inline testsuite/classlibrary/TestServerOptions.sc]: Please make unit tests deterministic -- random failures aren't fun!

Comment (telephon) [inline testsuite/classlibrary/TestServerOptions.sc]: ok, I'll do that.

**Comment by jamshark70** [comment]
FWIW, documentation on ContiguousBlockAllocator says (emphasis mine):

> *new (size, pos: 0)
> Create a new allocator with size slots. You may *block off* the first pos slots (the server's audioBusAllocator does this to reserve the hardware input and output buses).

"Block off" != "offset"

And this is consistent with the old PowerOfTwoAllocator behavior:

```
p = PowerOfTwoAllocator(8, 3);
{ p.alloc(1) }.dup(8);
-> [ 3, 4, 5, 6, 7, nil, nil, nil ]
```

Admittedly "size" is confusing as an argument name, but that confusion existed in PowerOfTwoAllocator as well, long before I wrote ContiguousBlockAllocator ;)

**Comment by telephon** [comment]
probably I accidentally looked up `ContiguousBlock`, so I didn't find it documented. Thanks for the pointers.

**Comment by telephon** [comment]
btw. `ContiguousBlockAllocator(3, 4)` breaks with an error. Should it be protected?

**Comment by jamshark70** [comment]
> probably I accidentally looked up ContiguousBlock, so I didn't find it documented.

Hm, worth documenting ContiguousBlock with a stub (simple class, doesn't need much -- but a link to ContiguousBlock would help).

> btw. ContiguousBlockAllocator(3, 4) breaks with an error. Should it be protected?

It's reasonable to throw an intelligible error here.

**Comment by jamshark70** [comment]
Incidentally -- if we're now potentially using large offsets to handle multiple clientIDs, it would be worth re-factoring ContiguousBlockAllocator at some point. Currently it creates a `size`-item array. If it's a high clientID, you might end up with an array of hundreds of `nil` items followed by a few dozen valid entries.

```
sc3> c = ContiguousBlockAllocator(1000, 900);
-> a ContiguousBlockAllocator
sc3> c.slotAt(\array).detectIndex(_.notNil);
-> 900  // that's right, 0..899 are nil (!)
```

The initial intent (again, cribbing from PowerOfTwoAllocator) seems to have been to "block off" a *small* number of low indices -- but we might not be using it only in that way anymore.

**Comment by telephon** [comment]
One way would be to replace `Array` by `Order`.

**Comment by telephon** [comment]
there will be a better more thorough PR by @adcxyz that revises the allocators.

**Comment by telephon** [comment]
check later if all issues are addressed by it.

**Pull request closed without merging**